pkg/cloudprovider/cloudprovider.go

```diff
@@ -200,7 +203,7 @@ func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *v1alpha5.
 	}
 	drifted, err := c.isNodeTemplateDrifted(ctx, machine, provisioner, nodeTemplate)
 	if err != nil {
-		return false, err
+		return false, cloudprovider.IgnoreMachineNotFoundError(err)
```

Comment:
I thought we wanted to move this [karpenter-core](https://github.com/aws/karpenter-core/pull/373),  https://github.com/aws/karpenter/pull/3949#discussion_r1218708224

Reply:
Ah good call, missed this in the rebase.